src/scripts/towns/BattleCafe.ts

```diff
@@ -0,0 +1,253 @@
+        }
+        if (BattleCafeController.isSpinning()) {
+            Notifier.notify({
+                message: 'Already spinning',
+                type: NotificationConstants.NotificationOption.danger,
+            });
+            return false;
+        }
+        if (BattleCafeController.spinsLeft() < 1) {
+            Notifier.notify({
+                message: 'No spins left',
+                type: NotificationConstants.NotificationOption.danger,
+            });
+            return false;
+        }
+        return BattleCafeController.getPrice(BattleCafeController.selectedSweet()).every(b => {
+            if (App.game.farming.berryList[b.berry]() < b.amount) {
+                Notifier.notify({
+                    message: 'Not enough berries',
+                    type: NotificationConstants.NotificationOption.danger,
+                });
+                return false;
+            }
+            return true;
+        });
+    }
+
+
+    private static getPrice(sweet: GameConstants.AlcremieSweet) : {berry: BerryType, amount: number}[] {
+        switch (sweet) {
+            case GameConstants.AlcremieSweet['Strawberry Sweet']:
+                return [
+                    {berry: BerryType.Cheri, amount: 500},
+                    {berry: BerryType.Leppa, amount: 400},
+                    {berry: BerryType.Razz, amount: 30},
+                ];
+            case GameConstants.AlcremieSweet['Love Sweet']:
+                return [
+                    {berry: BerryType.Roseli, amount: 20},
+                    {berry: BerryType.Liechi, amount: 10},
+                ];
+            case GameConstants.AlcremieSweet['Berry Sweet']:
+                return [
+                    {berry: BerryType.Yache, amount: 25},
+                    {berry: BerryType.Haban, amount: 25},
+                    {berry: BerryType.Wiki, amount: 150},
+                ];
+            case GameConstants.AlcremieSweet['Clover Sweet']:
+                return [
+                    {berry: BerryType.Wepear, amount: 30},
+                    {berry: BerryType.Aguav, amount: 30},
+                    {berry: BerryType.Lum, amount: 5},
+                ];
+            case GameConstants.AlcremieSweet['Flower Sweet']:
+                return [
+                    {berry: BerryType.Pomeg, amount: 30},
+                    {berry: BerryType.Nomel, amount: 30},
+                ];
+            case GameConstants.AlcremieSweet['Star Sweet']:
+                return [
+                    {berry: BerryType.Starf, amount: 3},
+                    {berry: BerryType.Lansat, amount: 3},
+                ];
+            case GameConstants.AlcremieSweet['Ribbon Sweet']:
+                return [
+                    {berry: BerryType.Chople, amount: 20},
+                    {berry: BerryType.Haban, amount: 20},
+                ];
+        }
+    }
+
+    private static buildEvolutions() : {[sweet: string] : {[spin: string]: PokemonItem}} {
```

Comment:
This looks like it returns the same data every time, can we just make it a static value instead of a function we call every spin?

Also I think we can avoid the double enum wrapping if you pass sweet/spin in directly instead of as `AlcremieSweet[sweet]` etc

Reply:
Yeah, i don't know how i ended up with that. That function had some iteration.
But i have fixed it now